Backend/routes/userRoutes.js

```
@@ -1,103 +1,116 @@
-      await newUser.save();
-      res.status(201).json({ message: "User created successfully", user: newUser });
-    } catch (error) {
-      console.error("❌ Error creating user:", error);
-      res.status(500).json({ message: "Internal server error" });
+            // 5. Save the User
+            await newUser.save();
+            res.status(201).json({ message: "User created successfully", user: newUser });
+        } catch (error) {
+            // 6. Error Handling
+            console.error("❌ Error creating user:", error);
+            res.status(500).json({ message: "Internal server error: " + error.message }); // Include error message for debugging
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)